### ISSUE TITLE
Dynamic finders are not returning promises on error

### ISSUE THREAD

**Issue by xedef** [balderdashy/waterline#673]
Let's say I have this code:
```
var name = req.param('name');

User.findOneByName(name)
.then(function(user){
    if (!user) {
        res.notFound();
    }
    else {
        res.json(user);
    }
})
.catch(function(err){
    res.badRequest('Something went wrong!');
});
```
If I don't provide the call with the name parameter, the dynamic finder will just throw an exception instead of going through ```.catch```. 
If I run the same code but calling ```findOne``` instead of a dynamic finder, the ```catch``` code is executed.

**Comment by devinivy** [comment]
Any luck with this?  Would be nice to see this resolved.

**Comment by sailsbot** [comment]
Thanks for posting, @xedef.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
